src/workerd/io/actor-sqlite.c++

```diff
@@ -379,6 +379,16 @@ void ActorSqlite::maybeDeleteDeferredAlarm() {
   }
 }
 
+void ActorSqlite::checkKeyValSize(Key& key, Value& value) {
+  auto limitLength = kv.getLimitLength() - 1;
+  JSG_REQUIRE(key.size() <= limitLength, RangeError,
+      kj::str("Keys cannot be larger than ", limitLength, " bytes. ", "A key of size ", key.size(),
+          " was provided."));
+  JSG_REQUIRE(value.size() <= limitLength, RangeError,
+      kj::str("Values cannot be larger than ", limitLength, " bytes. ", "A value of size ",
+          value.size(), " was provided."));
```

Comment:
This calculation is incorrect.  SQLITE_LIMIT_LENGTH is "[t]he maximum size of any string or BLOB or table row, in bytes."  The table row includes both the key and the value and some overhead.

You could make this more accurate by checking that the length of the key + the length of the value < the limit length, but it would still be wrong in a bunch of cases.

The problem that we have here is that if you `putMultiple` and some value is above the limit, KV-backed DOs will not commit anything while SQLite-backed DOs will still commit some values.  I thought we were going to put everything into a transaction and have SQLite do the check for us, and if the check failed, then we roll back the transaction.